### DISCUSSION TITLE
sites/siteid/prices response 16hr forecasts?

### DISCUSSION THREAD

**Discussion by AllTerrainAction** [amberelectric/public-api#25] [Q&A] [answered]
Not really an issue but more a question.. to understand why.

nemTime always ends at 4:00a no matter of the date range, I understand there is overlap when you widen date range out so there are never gaps in forecasts but how come it just doesn't do midnight to midnight? rather midnight to midnight +4 more hours ?

Even if you look at UTC time start and end, periods start UTC 4:00a but end last result as UTC 8:00a (giving 28hr forecasts?)

**Comment by madpilot** [comment, marked as answer]
I'm not seeing that... This is the last forecast interval I'm seeing:

```json
{
    "type": "ForecastInterval",
    "date": "2021-06-23",
    "duration": 30,
    "startTime": "2021-06-23T03:30:01Z",
    "endTime": "2021-06-23T04:00:00Z",
    "nemTime": "2021-06-24T00:00:00+10:00",
    "perKwh": 18.98369,
    "renewables": 28.600951950021653,
    "spotPerKwh": 6.819695300000001,
    "channelType": "general",
    "spikeStatus": "none",
    "range": {
      "min": 4.35601,
      "max": 26.93593
    }
  }
  ```
  
  If you can send me your Site ID and the query params you are using, I can have a look.

**Reply by AllTerrainAction** [reply]
hmm,

I always end up no matter which start or end date with something ending in..

used https://api.amber.com.au/v1/sites/01EGFXYCSC05F7MJDAX4RHBAZD/prices?startDate=2021-06-23&endDate=2021-06-25&resolution=30

below is last array in response body

        (
            [type] => ForecastInterval
            [date] => 2021-06-24
            [duration] => 30
            [startTime] => 2021-06-23T07:30:01Z
            [endTime] => 2021-06-23T08:00:00Z
            **[nemTime] => 2021-06-24T04:00:00+10:00**
            [perKwh] => 15.05677
            [renewables] => 29.486391539888
            [spotPerKwh] => 2.7414673
            [channelType] => general
            [spikeStatus] => none
            [range] => stdClass Object
                (
                    [min] => 1.81599
                    [max] => 4.44444
                )

        )

**Reply by madpilot** [reply]
Ah. You have tomorrow as the end date - the forecast doesn't go all the way to midnight the next day. We only have ~16 hours of forecasts to work off. If you try again after 12:01, it should stretch out to nemTime `2021-06-24T04:30:00+10:00`

**Reply by AllTerrainAction** [reply]
> Ah. You have tomorrow as the end date - the forecast doesn't go all the way to midnight the next day. We only have ~16 hours of forecasts to work off. If you try again after 12:01, it should stretch out to nemTime `2021-06-24T04:30:00+10:00`

okay so this makes sense! limits on forcasts following day. Thanks - sorry!

**Reply by AllTerrainAction** [reply]
@madpilot ... okay something strange just happened - did we just get an update to make it work the way I was hoping lol? 

![image](https://user-images.githubusercontent.com/36219843/123026566-bfb02980-d41f-11eb-9964-3eb34353edb8.png)

**Reply by madpilot** [reply]
Yeah, looks like we now have data all the way up until 4am on the 25th. AEMO must have re-run their prediction algo and extended the forecast. Looks like the new results came through sometime between 12 and 12:30.
